flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeTypeSerializerConfigSnapshot.java

```diff
@@ -110,4 +123,20 @@ public boolean equals(Object obj) {
 	public int hashCode() {
 		return nestedSerializersAndConfigs.hashCode();
 	}
+
+	private Map<TypeSerializer<?>, Integer> buildSerializerIndices() {
```

Comment:
I see that the concrete implementation is always an identity hash map. since it behaves different from what you expect by a normal map (using equals/hashcode) and in that sense violates LSP, I suggest you wrap it in an own class called `SerializerIndex`.